apollo-federation/src/sources/connect/spec/type_and_directive_specifications.rs

```diff
@@ -229,6 +232,41 @@ pub(super) fn check_or_add(
         type_name: connect_batch.name.clone(),
     };
 
+    // @connect error settings
+
+    let connect_errors_field_list = vec![
+        InputValueDefinition {
+            description: None,
+            name: name!(message),
+            ty: Type::Named(json_selection_spec.name.clone()).into(),
+            default_value: None,
+            directives: Default::default(),
+        },
+        InputValueDefinition {
+            description: None,
+            name: name!(extensions),
+            ty: Type::Named(json_selection_spec.name.clone()).into(),
+            default_value: None,
+            directives: Default::default(),
+        },
+    ];
+
+    let mut connect_errors_fields = IndexMap::with_hasher(Default::default());
+    for field in connect_errors_field_list {
+        connect_errors_fields.insert(field.name.clone(), Component::new(field));
+    }
```

Comment:
We can definitely write this without making a vec and then an IndexMap. Here's one idea:

```suggestion
    let connect_errors_fields = [
        ERRORS_MESSAGE_ARGUMENT_NAME,
        ERRORS_EXTENSIONS_ARGUMENT_NAME,
    ]
    .into_iter()
    .map(|name| {
        (
            name.clone(),
            Component::new(InputValueDefinition {
                description: None,
                name,
                ty: Type::Named(json_selection_spec.name.clone()).into(),
                default_value: None,
                directives: Default::default(),
            }),
        )
    })
    .collect();
```

Though I wonder why we don't add descriptions to these...

Reply:
So this code I made identical to what was done previously (See: http_header_mapping_field_list, connect_http_field_list).

Similar to previous comments... I want to be consistent so if we want to update all of these, I'd prefer to do in a follow on PR instead of having my code be a special snowflake. ❄️ 😆